### ISSUE TITLE
Return Location header on resource creation or update

### ISSUE THREAD

**Issue by dbutenhof** [distributed-system-analysis/pbench#2242]
A widespread REST convention calls for reporting the URI at which a resource may be queried by setting the `Location` header in the HTTP response. We should do this for create and update operations.

In some cases, e.g., for a user, the URI is obvious (`Location: <host>/api/v1/users/<name>`); for others it may not be so obvious, and we need to figure out for example how we report an uploaded dataset. I.e., do we report the path to the tarball, or to something else? (In our current staged design, an uploaded tarball won't be unpacked or indexed for "a while" subsequent to upload, so there may be nothing else we _can_ point to, but this won't always be the case.)

**Comment by webbnh** [comment]
> In our current staged design, an uploaded tarball won't be unpacked or indexed for "a while" subsequent to upload, so there may be nothing else we can point to, but this won't always be the case.

Yes, but...if we know where it will be unpacked _to_ (that is, if we know what the URI for accessing the data will be once it is available), we could return that, and the client could then poll it as a means to know when the processing is done.

**Comment by dbutenhof** [comment]
There are better precedents for reporting and monitoring asynchronous operations, if that's how we want to represent it. We accepted and completed a request to "create a resource" and we're returning a link to that resource; if it doesn't yet exist, that's at best a bit of a cheat, and requires polling by the client to figure out when it actually does exist.

Right now I don't want to get into even thinking about managing asynchronous client-visible tasks in the server, although I suspect at some point we'll want that. But at that point, we could report the upload as a 202 with a response header link to a task URI that could be monitored (ideally with a mechanism for the client to subscribe to task updates so they don't have to poll).

Initially, though, I would just want to return a `Location` header that points to something that exists. Maybe it's just the tarball itself on the server; that's not very interesting, but it's concrete. (And again, at some point, a task/event system that would let the client follow the tarball through our component states...)

**Comment by webbnh** [comment]
My concern is that you feel obliged to return a `Location` header and you're now casting about for something to put in it.  Before we put a URL for the tarball there, I would want to see some indication that someone actually wants that information.  If not, doing so is worse than useless because it likely creates a constraint on future implementations.  (Do we really want to be a file server for tarballs?...or, is the tarball just a transfer medium, and what we want to be in the business of is serving "data", whatever that ends up being?)

Now that you mention it, returning a `202` response looks like exactly the right thing to do:  the server accepts the data, the result is a "success", and the data will become available "later".

**Comment by portante** [comment]
We've decided we are not going to pursue this further.
